packages/pipeline/test/pdk-pipeline.test.ts

```diff
@@ -195,4 +195,191 @@ describe("PDK Pipeline Unit Tests", () => {
       "Stage/AppStack/Non-Compliant/Resource"
     );
   });
+
+  it("Feature Branches", () => {
+    delete process.env.BRANCH;
+    const app = PDKNag.app();
+    const stack = new Stack(app);
+
+    const pipeline = new PDKPipeline(stack, "FeatureBranches", {
+      primarySynthDirectory: "cdk.out",
+      repositoryName: "FeatureBranches",
+      defaultBranchName: "mainline",
+      branchNamePrefixes: [""],
+      synth: {},
+      crossAccountKeys: false,
+    });
+
+    const stage = new Stage(app, "Stage");
+    const appStack = new Stack(stage, "AppStack");
+    new Asset(appStack, "Asset", {
+      path: path.join(__dirname, "pdk-pipeline.test.ts"),
+    });
+
+    pipeline.addStage(stage);
+    pipeline.buildPipeline();
+
+    app.synth();
+    expect(app.nagResults().length).toEqual(0);
+    expect(Template.fromStack(stack)).toMatchSnapshot();
+  });
+
+  it("Feature Branches - mainline", () => {
+    process.env.BRANCH = "mainline";
+    const app = PDKNag.app();
+    const stack = new Stack(app);
+    const branchPrefix = "";
+
+    const pipeline = new PDKPipeline(stack, branchPrefix + "FeatureBranches", {
+      primarySynthDirectory: "cdk.out",
+      repositoryName: "FeatureBranches",
+      defaultBranchName: "mainline",
+      branchNamePrefixes: [""],
+      synth: {},
+      crossAccountKeys: false,
+    });
+
+    const stage = new Stage(app, branchPrefix + "Stage");
+    const appStack = new Stack(stage, "AppStack");
+    new Asset(appStack, "Asset", {
+      path: path.join(__dirname, "pdk-pipeline.test.ts"),
+    });
+
+    pipeline.addStage(stage);
+    pipeline.buildPipeline();
+
+    app.synth();
+    expect(app.nagResults().length).toEqual(0);
+    expect(Template.fromStack(stack)).toMatchSnapshot();
+    delete process.env.BRANCH;
```

Comment:
It might be worth cleaning up the environment in a `finally` block to reduce the chances of a failure in one test affecting another :)

Reply:
I implemented an `afterEach` function to restore `process.env`.